### PR TITLE
8311247: Some cpp files are compiled with -std:c11 flag

### PR DESCRIPTION
**Pull request by djelinski**
Please review this patch that configures C++ arguments on build jobs that involve compiling CPP files. As a result of this change, CPP files are compiled with `-std:c++14` command line argument instead of `-std:c11`, which is used when C++ arguments are not configured.
While at it, I simplified the `java.security.jgss/Lib.gmk` file by moving the additional include directory to `EXTRA_HEADER_DIRS`.

This patch fixes the following clang warning:
```
warning: argument unused during compilation: '-std:c11'
```

Microsoft states that [std:c++14 is the default](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170), so there shouldn't be any differences in produced code.

Testing:
- verified that after the changes, all CPP files are compiled with `std:c++14` instead of `-std:c11`
- spot-checked a few `cmdline` files that changed after this patch was applied; the `-std` change was the only difference
- tier1-5 builds, tier1-2 tests and client libs tests continue to pass on Windows, Linux & MacOS.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8311247](https://bugs.openjdk.org/browse/JDK-8311247): Some cpp files are compiled with -std:c11 flag (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Julian Waters](https://openjdk.org/census#jwaters) (@TheShermanTanker - Committer)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14758/head:pull/14758` \
`$ git checkout pull/14758`

Update a local copy of the PR: \
`$ git checkout pull/14758` \
`$ git pull https://git.openjdk.org/jdk.git pull/14758/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14758`

View PR using the GUI difftool: \
`$ git pr show -t 14758`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14758.diff">https://git.openjdk.org/jdk/pull/14758.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14758#issuecomment-1619000429)

### CODE REVIEW

File: make/modules/java.desktop/lib/Awt2dLibraries.gmk
Comment (aivanov-jdk) [inline make/modules/java.desktop/lib/Awt2dLibraries.gmk:135]: `LIBAWT_CFLAGS` applies to both C and C++, right? It contains some common flags; it follows the pattern with other libs below.

Comment (djelinski) [reply make/modules/java.desktop/lib/Awt2dLibraries.gmk:135]: That's correct. `LIBAWT_CFLAGS contains a bunch of `-D` flags, which are common for C and C++, and `-EHsc` flag, which is probably C++ only, but the compiler doesn't complain.

`jabswitch` below is using some C++-specific flags, but it only compiles one C++ file and no C files. I suppose we could simplify that makefile at some point. All other flags are common between C and C++.

Comment (aivanov-jdk) [reply make/modules/java.desktop/lib/Awt2dLibraries.gmk:135]: Thank for clarification.

**Comment by aivanov-jdk** [comment]
/reviewers 2

I believe a reviewer from the build team is still required.

**Review by aivanov-jdk — APPROVED** [review]

**Comment by TheShermanTanker** [comment]
Looks good to me. I actually have a similar change applied locally on my repo too

**Review by TheShermanTanker — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Comment by mrserb** [comment]
Did we document the current standard of C++ and C we can use?

**Comment by TheShermanTanker** [comment]
@mrserb We use C11 and C++14, and we're soon to move to C++17 once the AIX port gets a compiler that can accept it

**Comment by djelinski** [comment]
The C++ standard is only documented for Hotspot, and the C standard is not documented anywhere as far as I could tell.

(Still waiting for a review from build-dev)

**Review by erikj79 — APPROVED** [review]
I'm ok with this, but it's unfortunate that we need to repeat so much.

**Comment by djelinski** [comment]
> it's unfortunate that we need to repeat so much.

Right. The default make convention uses `CPPFLAGS` for flags shared between preprocessor and both C and C++ compilers. We don't have any shared variable as far as I could tell.
Thanks for the reviews!

/integrate

**Pull request closed without merging**
